src/consensus/validator.rs

```diff
@@ -54,12 +54,12 @@ impl ShardValidator {
     }
 
     pub fn get_current_height(&self) -> Result<u64, BlockProposerError> {
-        match &self.block_proposer {
-            None => Ok(0),
-            Some(block_proposer) => block_proposer
-                .block_store
-                .max_block_number(self.shard_id.shard_id())
-                .map_err(|err| BlockProposerError::BlockStorageError(err)),
+        if let Some(proposer) = &self.block_proposer {
```

Comment:
nit: this took me a lot longer to read than I might have expected (maybe I'm still too new to rust)

```rust
pub fn get_current_height(&self) -> Result<u64, BlockProposerError> {
    if let Some(p) = &self.block_proposer {
        return Ok(p.get_confirmed_height().block_number);
    }
    if let Some(p) = &self.shard_proposer {
        return Ok(p.get_confirmed_height().block_number);
    }
    Err(BlockProposerError::NoProposerSet)
}
```

I like return above else if / else.

I like `p` here, proposer is just too close to both block_proposer and shard_proposer, with so many `proposer`s it took a long time to realize the two cases were actually different. Instead of `p` I think `block_proposer` and `shard_proposer` could work too.

Feel free to ignore any of above.

Reply:
Fixed, but also removed the error. If this fails, there's no recovering, consensus is going to halt, and the error is not handled anyway, so might as well panic.